DataFormats/Scouting/interface/ScoutingParticle.h

```diff
@@ -10,10 +10,10 @@ class ScoutingParticle
     public:
         //constructor with values for all data fields
         ScoutingParticle(float pt, float eta, float phi, float m, 
-                int pdgId):
-            pt_(pt), eta_(eta), phi_(phi), m_(m), pdgId_(pdgId) {}
+		 int pdgId, int vertex):
+            pt_(pt), eta_(eta), phi_(phi), m_(m), pdgId_(pdgId), vertex_(vertex) {}
         //default constructor
-        ScoutingParticle():pt_(0), eta_(0), phi_(0), m_(0), pdgId_(0) {}
+        ScoutingParticle():pt_(0), eta_(0), phi_(0), m_(0), pdgId_(0), vertex_(0) {}
```

Comment:
Given 0 is a valid index, how about using max int instead? That way reading old data one can spot if the variable is not filled.